pkg/config/jobs.go

```diff
@@ -534,8 +536,25 @@ func NewGitHubDeferredChangedFilesProvider(client githubClient, org, repo string
 			if err != nil {
 				return nil, fmt.Errorf("error getting pull request changes: %w", err)
 			}
-			for _, change := range changes {
-				changedFiles = append(changedFiles, change.Filename)
+
+			// Fallback to use gitClient since github API truncated the response
+			if len(changes) == github.ChangesFilesLimit && gc != nil {
```

Comment:
As suggested here: https://github.com/kubernetes/test-infra/pull/30615#discussion_r1339994398 this is now a fallback, that is only executed when:
* gitclient is not nil
* length of changes is exactly 3000

Of course, this means that if unluckily number of changes is exactly 3000, we would run the git client flow for nothing.